trillian/ctfe/handlers.go

```diff
@@ -230,32 +230,40 @@ type logInfo struct {
 	validationOpts CertValidationOpts
 	// rpcClient is the client used to communicate with the trillian backend
 	rpcClient trillian.TrillianLogClient
-	// signer signs objects
+	// signer signs objects (e.g. STHs, SCTs) for regular logs
 	signer crypto.Signer
-
-	// Cache the last signature generated for an STH, to reduce re-signing and
-	// slightly reduce the chances of being able to fingerprint get-sth users by
-	// their STH signature value.
-	sthSigCache SignatureCache
+	// sthGetter provides STHs for the log
+	sthGetter STHGetter
 }
 
 // newLogInfo creates a new instance of logInfo.
-func newLogInfo(logID int64, prefix string, validationOpts CertValidationOpts, rpcClient trillian.TrillianLogClient, signer crypto.Signer, instanceOpts InstanceOptions, timeSource util.TimeSource) *logInfo {
-	ctx := &logInfo{
+func newLogInfo(
+	logID int64, prefix string, isMirror bool, validationOpts CertValidationOpts,
+	rpcClient trillian.TrillianLogClient, signer crypto.Signer,
+	instanceOpts InstanceOptions, timeSource util.TimeSource,
+) *logInfo {
+	li := &logInfo{
 		logID:          logID,
 		urlPrefix:      prefix,
 		LogPrefix:      fmt.Sprintf("%s{%d}", prefix, logID),
+		isMirror:       isMirror,
 		rpcClient:      rpcClient,
 		signer:         signer,
 		TimeSource:     timeSource,
 		instanceOpts:   instanceOpts,
 		validationOpts: validationOpts,
 		RequestLog:     instanceOpts.RequestLog,
 	}
+	li.sthGetter = &LogSTHGetter{li: li}
```

Comment:
I don't like having to create both of them but Go makes this awkward.

Reply:
Made this 1 line longer.